.github/workflows/lint.yaml

```diff
@@ -6,21 +6,30 @@ on:
 
 name: lint
 
+env:
+  PYTHONUNBUFFERED: "1"
+  UV_SYSTEM_PYTHON: 1
+
 jobs:
   lint-ruff:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Setup python
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+          cache-dependency-glob: pyproject.toml
+
+      - name: Setup Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.x"
-          cache: "pip"
+          python-version-file: pyproject.toml
```

Comment:
This just install the min version specified in our `pyproject.toml`, in this case `3.10`.